client/blocks/site-address-changer/index.jsx

```diff
@@ -70,8 +76,23 @@ export class SiteAddressChanger extends Component {
 			domainFieldValue,
 			newDomainSuffix.substr( 1 ),
 			oldDomain,
-			type
+			type,
+			option === 'discard',
+			option !== 'redirect'
```

Comment:
If we want to change the site address before adding the redirect to the cart and going to checkout (I'm not sure if that's the right order!), there should be some `await` or `.then` here to wait for the address change REST request to finish before proceeding.